playbooks/openstack/sample-inventory/inventory.py

```diff
@@ -98,8 +98,45 @@ def build_inventory():
             hostvars['openshift_node_labels'] = node_labels
 
         inventory['_meta']['hostvars'][server.name] = hostvars
+
+    kuryr_vars = _get_kuryr_vars(cloud)
+    if kuryr_vars:
+        inventory['OSEv3']['vars'] = kuryr_vars
+
     return inventory
 
 
+def _get_kuryr_vars(cloud_client):
+    """Returns a dictionary of Kuryr variables resulting of heat stacking"""
+    # TODO: Filter the cluster stack with tags once it is supported in shade
+    hardcoded_cluster_name = 'openshift.example.com'
+    stack = cloud_client.get_stack(hardcoded_cluster_name)
+    if stack is None or stack['stack_status'] != 'CREATE_COMPLETE':
+        return None
+
+    data = {}
+    for output in stack['outputs']:
```

Comment:
could we instead use metadata properties, like we do above, and omit the stack outputs parsing, and hardocded stack names as well?

Reply:
We're working on not having to hardcode the stack name (the idea is adding a tag to the stack which we'll filter for here).

But this is exactly what Heat outputs are for? Why shouldn't we use them? Also, where should we attach this metadata other than the stack? Some of the data here (the network and SG ids) are not known before the stack is created.

Reply:
This is what heat outputs are for